### PR TITLE
Install torchdynamo as part of most CI jobs

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #79099
* __->__ #79051
* #79042

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: .jenkins/pytorch/common_utils.sh [hunk 1]
```diff
@@ -106,3 +106,7 @@ function clone_pytorch_xla() {
     popd
   fi
 }
+
+function install_torchdynamo() {
+  pip_install --user "git+https://github.com/pytorch/torchdynamo.git@$(cat .github/torchdynamo_commit_hash.txt)"
```
Comment (jansel) [inline .jenkins/pytorch/common_utils.sh:111]: Where does `.github/torchdynamo_commit_hash.txt` come from?

Comment (ezyang) [reply .jenkins/pytorch/common_utils.sh:111]: this: https://github.com/pytorch/pytorch/pull/79042 it's hard coded and gets a PR to be updated nightly

**Review by jansel — APPROVED** [review]

**Review by anijain2305 — APPROVED** [review]

**Pull request closed without merging**